efaar_benchmarking/benchmarking.py

```diff
@@ -13,38 +15,70 @@
 )
 
 
-def pert_stats(
-    map_data: Bunch,
-    pert_pval_thr: float = cst.PERT_SIG_PVAL_THR,
-):
+def univariate_cons_metric(arr: np.ndarray, null=None) -> tuple[float, float]:
     """
-    Calculate perturbation statistics based on the provided map data.
+    Calculate the univariate consistency metric, i.e. average cosine angle and associated p-value, for a given array.
 
     Args:
-        map_data (Bunch): Map data containing metadata.
-        pert_pval_thr (float): Perturbation significance threshold. Defaults to cst.PERT_SIG_PVAL_THR.
+        arr (numpy.ndarray): The input array.
+        null (numpy.ndarray, optional): Null distribution of the metric. Default is None.
 
     Returns:
-        dict: Dictionary containing perturbation statistics:
-            - "all_pert_count": Total count of perturbations.
-            - "pp_pert_count": Count of perturbations that meet the significance threshold.
-            - "pp_pert_percent": Percentage of perturbations that meet the significance threshold.
+        tuple: A tuple containing the average angle (avg_angle) and p-value (pval) of the metric.
+           If the length of the input array is less than 3, returns (None, None).
+           If null is None, returns (avg_angle, None).
     """
+    if len(arr) < 3:
+        return None, None
+    cosine_sim = cosine_similarity(arr)
+    avg_angle = np.arccos(cosine_sim[np.tril_indices(cosine_sim.shape[0], k=-1)]).mean()
+    if null is None:
+        return avg_angle, None
+    else:
+        sorted_null = np.sort(null[len(arr)])
```

Comment:
question: Why is the null being subsetted here?

Reply:
I see why now after going through the rest of the PR, null is a dictionary, not an ndarray.  The doc string should be updated, but I think it could make more sense to refactor this slightly, so this function takes the np.array and the correct null, instead of having the null logic int this function.